sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -324,6 +328,41 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
     }
   }
 
+  // Wait for command spawned via Process
+  private def waitForProcess(cmd: Process, durationMs: Long): Option[Int] = {
+    val endTime = System.currentTimeMillis() + durationMs
+    do {
+      Thread.sleep(10)
+      if (!cmd.isAlive()) {
+        return Some(cmd.exitValue())
+      }
+    } while (System.currentTimeMillis() < endTime)
+    None // Timed out
+  }
+
+  // Try to run nvidia-smi when task fails due to a cuda exception.
+  private def logGpuDebugInfoAndExit(systemExitCode: Int) = synchronized {
+    try {
+      val nvidiaSmiStdout = new StringBuilder
+      val nvidiaSmiStderr = new StringBuilder
+      val cmd = "nvidia-smi".run(
+        ProcessLogger(s => nvidiaSmiStdout.append(s + "\n"), s => nvidiaSmiStderr.append(s + "\n")))
+      waitForProcess(cmd, 2000) match {
+        case Some(exitStatus) =>
+          if (exitStatus == 0) {
+            logWarning("nvidia-smi:\n" + nvidiaSmiStdout)
+          } else {
+            logWarning("nvidia-smi failed with: " + nvidiaSmiStdout + nvidiaSmiStderr)
+          }
+        case None => logWarning("nvidia-smi command timed out")
+      }
+    } catch {
+      case e: Throwable => logWarning("nvidia-smi process failed with: "
```

Comment:
perhaps move `logWarning` to the next line? minor nit on indentation, so we can probably just interpolate `${e.getMessage()}`